aws-ssm-parameter/src/main/java/com/amazonaws/ssm/parameter/TagHelper.java

```diff
@@ -0,0 +1,278 @@
+     */
+    public static Set<String> generateTagsToRemove(final Map<String, String> previousTags, final Map<String, String> desiredTags) {
+        final Set<String> desiredTagNames = desiredTags.keySet();
+
+        return previousTags.keySet().stream()
+            .filter(tagName -> !desiredTagNames.contains(tagName))
+            .collect(Collectors.toSet());
+    }
+
+    /**
+     * generateTagsToAdd
+     *
+     * Determines the tags the customer desired to define or redefine.
+     */
+    public static Set<Tag> generateTagsToAdd(final Set<Tag> previousTags, final Set<Tag> desiredTags) {
+        return Sets.difference(new HashSet<>(desiredTags), new HashSet<>(previousTags));
+    }
+
+    /**
+     * getTagsToRemove
+     *
+     * Determines the tags the customer desired to remove from the function.
+     */
+    public static Set<Tag> generateTagsToRemove(final Set<Tag> previousTags, final Set<Tag> desiredTags) {
+        return Sets.difference(new HashSet<>(previousTags), new HashSet<>(desiredTags));
+    }
+
+
+    /**
+     * tagResource during update
+     *
+     * Calls the service:TagResource API.
+     */
+    public ProgressEvent<ResourceModel, CallbackContext>
+    tagResource(final AmazonWebServicesClientProxy proxy, final ProxyClient<SsmClient> serviceClient, final ResourceModel resourceModel,
+                final ResourceHandlerRequest<ResourceModel> handlerRequest, final CallbackContext callbackContext, final Map<String, String> addedTags, final Logger logger) {
+        logger.log(String.format("[UPDATE][IN PROGRESS] Going to add tags for ... resource: %s with AccountId: %s",
+        resourceModel.getName(), handlerRequest.getAwsAccountId()));
+        List<Tag> tagsToAdd = convertToList(convertToSet(addedTags));
+        return proxy.initiate("AWS-SSM-Parameter::TagOps", serviceClient, resourceModel, callbackContext)
+            .translateToServiceRequest(model ->
+                Translator.tagResourceRequest(model, tagsToAdd))
+            .makeServiceCall(this::addResourceTag)
+            .progress();
+    }
+
+    /**
+     * untagResource during update
+     *
+     * Calls the service:UntagResource API.
+     */
+    public ProgressEvent<ResourceModel, CallbackContext>
+    untagResource(final AmazonWebServicesClientProxy proxy, final ProxyClient<SsmClient> serviceClient, final ResourceModel resourceModel,
+                  final ResourceHandlerRequest<ResourceModel> handlerRequest, final CallbackContext callbackContext, final Set<String> removedTags, final Logger logger) {
+        logger.log(String.format("[UPDATE][IN PROGRESS] Going to remove tags for ... resource: %s with AccountId: %s",
+        resourceModel.getName(), handlerRequest.getAwsAccountId()));
+        List<String> tagsToRemove = convertToList(removedTags);
+        return proxy.initiate("AWS-SSM-Parameter::TagOps", serviceClient, resourceModel, callbackContext)
```

Comment:
nit: change call-graph name. Same call-graph name is being used for tag and "un-tag" operations. This can be helpful for debugging purposes.